src/geeknn/ordination/utils.py

```diff
@@ -1,17 +1,24 @@
-from typing import Any
+from __future__ import annotations
 
 import ee
 from pydantic import BaseModel, ConfigDict
 
 
 class Colocation(BaseModel):
```

Comment:
I definitely understand what you're saying here, but I'm struggling with one aspect of this.  Setting up this crosswalk requires the user to specify the feature collection that holds the crosswalk (as well as the fields involved).  Where would the user specify the feature collection if this class was internal?  I think I'm totally missing something.

Reply:
I'm still pretty fuzzy on this part of the workflow, so any confusion is definitely coming from me! I had assumed that the typical usage would be to have one feature collection with all of the species, environmental, plot ID, and optional location ID fields, and you would need to feed that into `Colocation` to get a new `colocation_obj`, then feed both into `predict`, which seemed slightly redundant. After taking a closer look at the tests, I see that you're bringing in a separate collection for the plot/location ID crosswalk. I take it that's a more typical workflow?

Would it be possible/preferable to require the user to pre-join the location ID field into the main `fc` if they want to filter co-located plots? It would add another step up-front (maybe there could be a helper func, since joins in EE are admittedly a hassle), but overall maybe it would simplify things by hiding the colocation details and forcing the user to just deal with one FC and field names?

You have a much better idea of the typical workflow than I do, so take any of my input here with a grain of salt!

Reply:
> I had assumed that the typical usage would be to have one feature collection with all of the species, environmental, plot ID, and optional location ID fields, and you would need to feed that into `Colocation` to get a new `colocation_obj`, then feed both into `predict`, which seemed slightly redundant. After taking a closer look at the tests, I see that you're bringing in a separate collection for the plot/location ID crosswalk. I take it that's a more typical workflow?

Ah, that makes a lot more sense and it was likely my bad explanation or lack of details that led to the confusion.  If we step back to stock GNN mapping for NWFP, I have a single file that crosswalks plot IDs to location IDs and this file can be used across all modeling regions so that we don't actually add the location ID to the files we use in modeling.  So, again because of historical patterns, I've always imagined this to be a separate file that isn't specifically tied to modeling.

> Would it be possible/preferable to require the user to pre-join the location ID field into the main fc if they want to filter co-located plots? It would add another step up-front (maybe there could be a helper func, since joins in EE are admittedly a hassle), but overall maybe it would simplify things by hiding the colocation details and forcing the user to just deal with one FC and field names?

This seems like a reasonable way to handle it.  These types of decisions will come into play for `sknnr` (or the wrapper package around `sknnr` that works with AA) as well.  This would still allow the user to have a main lookup feature collection, but not have the user be responsible for instantiating the object.  

Having said this and in the interest of time, I think this one feels like it requires a bit more thought on how to best implement this, so I'm going to punt on this and create a new issue to deal with it.  So we'll still have the user-managed `Colocation` class in this PR, but hopefully remedied soon.

Reply:
Sounds like a good plan!